operators/audio/audio_decoder.hpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "ocos.h"
+
+#include <list>
+#define DR_FLAC_IMPLEMENTATION
+#include "dr_flac.h"
+#define DR_MP3_IMPLEMENTATION 1
+#define DR_MP3_FLOAT_OUTPUT 1
+#include "dr_mp3.h"
+#define DR_WAV_IMPLEMENTATION
+#include "dr_wav.h"
+
+#include <gsl/util>
+#include "string_utils.h"
+#include "string_tensor.h"
+
+struct KernelAudioDecoder : public BaseKernel {
+ public:
+  KernelAudioDecoder(const OrtApi& api, const OrtKernelInfo& info) : BaseKernel(api, info) {
+  }
+
+  enum {
+    DEFAULT_STREAM = 0,
+    WAV_STREAM,
+    MP3_STREAM,
+    FLAC_STREAM
+  };
+
+  int ReadStreamFormat(OrtKernelContext* context, const int64_t* p_data) {
+    int stream_format = DEFAULT_STREAM;
+    const OrtValue* ov_format = ort_.KernelContext_GetInput(context, 1);
+    if (ov_format != nullptr) {
+      std::vector<std::string> str_format;
+      GetTensorMutableDataString(api_, ort_, context, ov_format, str_format);
+      const char* formats[] = {"default",
+                               "wav",
+                               "mp3",
+                               "flac"};
+      for (auto fmt : formats) {
+        if (str_format[0] == fmt) {
+          break;
+        }
+        stream_format++;
```

Comment:
I think that using an enumeration type conveys more meaning than an int. this effectively maps from a string value to the enumeration value and doesn't have to be implemented with an integer increment.

Reply:
not sure you are suggesting there is any new C++ feature that get type string from enum class. But I just noticed we could static initialize a std::map now, which makes the code look more modern C++ style now.

Reply:
> not sure if you are suggesting there is any new C++ feature that get type string from enum class

I'm not aware of this, it would be nice though.

but yeah, I was thinking of a map lookup. actually with the small number of entries we can probably get by with scanning an array of pairs too, but map is ok.